packages/flutter/lib/src/widgets/window.dart

```diff
@@ -2,5 +2,307 @@
+    if (!owner.hasTopLevelWindows()) {
+      // No more top-level windows, exit the application.
+      ServicesBinding.instance.exitApplication(AppExitType.cancelable);
+    }
+  }
+}
+
+/// A controller for a regular window.
+///
+/// A regular window is a traditional window that can be resized, minimized,
+/// maximized, and closed. Upon construction, the window is created for the
+/// platform with the provided properties.
+///
+/// This class does not interact with the widget tree. Instead, it is typically
+/// provided to the [RegularWindow] widget, who does the work of rendering the
+/// content inside of this window.
+///
+/// An example usage might look like:
+/// ```dart
+/// final RegularWindowController controller = RegularWindowController(
+///   size: const Size(800, 600),
+///   sizeConstraints: const BoxConstraints(minWidth: 640, minHeight: 480),
+///   title: "Example Window",
+/// );
+/// runWidget(RegularWindow(
+///   controller: controller,
+///   child: MaterialApp(home: Container())));
+/// ```
+///
+/// When provided to a [RegularWindow] widget, widgets inside of the [child]
+/// parameter will have access to the [RegularWindowController] via the
+/// [WindowControllerContext] widget.
+abstract class RegularWindowController extends WindowController {
+  /// Creates a [RegularWindowController] with the provided properties.
+  /// Upon construction, the window is created for the platform.
+  ///
+  /// [contentSize] Initial content size of the window.
+  /// [title] the title of the window
+  /// [state] the initial state of the window
+  /// [delegate] optional delegate for the controller controller.
+  factory RegularWindowController({
+    required WindowSizing contentSize,
+    String? title,
+    WindowState? state,
+    RegularWindowControllerDelegate? delegate,
+  }) {
+    WidgetsFlutterBinding.ensureInitialized();
+    final WindowingOwner owner = WidgetsBinding.instance.windowingOwner;
+    final RegularWindowController controller = owner.createRegularWindowController(
+      contentSize: contentSize,
+      delegate: delegate ?? RegularWindowControllerDelegate(),
+    );
+    if (title != null) {
+      controller.setTitle(title);
+    }
+    if (state != null) {
+      controller.setState(state);
+    }
+    return controller;
+  }
+
+  @protected
+  /// Creates an empty [RegularWindowController].
+  RegularWindowController.empty();
+
+  @override
+  WindowArchetype get type => WindowArchetype.regular;
+
+  /// The current state of the window.
+  WindowState get state;
+
+  /// Request change for the window content size.
+  ///
+  /// [contentSize] describes the new requested window size. The properties
+  /// of this object are applied independently of each other. For example,
+  /// setting [WindowSizing.size] does not affect the [WindowSizing.constraints]
+  /// set previously.
+  ///
+  /// System compositor is free to ignore the request.
+  void setContentSize(WindowSizing contentSize);
```

Comment:
Should it be `updateContentSize`, then?

Or even, make the property `WindowSizing sizing;`?

Reply:
I don't mind. Will change it.